core/arch/arm/plat-stm/main.c

```diff
@@ -115,3 +131,132 @@ vaddr_t pl310_base(void)
 	}
 	return PL310_BASE;
 }
+
+void arm_cl2_config(vaddr_t pl310);
+void arm_cl2_config(vaddr_t pl310)
+{
+	uint32_t v;
+	/* pl310 off */
+	write32(0, pl310 + PL310_CTRL);
+
+	/*
+	 * TAG RAM Control Register
+	 *
+	 * bit[10:8]:1 - 2 cycle of write accesses latency
+	 * bit[6:4]:1 - 2 cycle of read accesses latency
+	 * bit[2:0]:1 - 2 cycle of setup latency
+	 */
+	v = read32(pl310 + PL310_TAG_RAM_CTRL);
+	v &= 0xFFFFF888;
+	v |= 0xFFFFF999;
+	write32(v, pl310 + PL310_TAG_RAM_CTRL);
+
+	/*
+	 * DATA RAM Control Register
+	 *
+	 * bit[10:8]:2 - 3 cycle of write accesses latency
+	 * bit[6:4]:2 - 3 cycle of read accesses latency
+	 * bit[2:0]:2 - 3 cycle of setup latency
+	 */
+	v = read32(pl310 + PL310_DATA_RAM_CTRL);
+	v &= 0xFFFFF888;
+	v |= 0xFFFFFAAA;
+	write32(v, pl310 + PL310_DATA_RAM_CTRL);
+
+	/*
+	 * Auxiliary Control Register
+	 *
+	 * I/Dcache prefetch enabled (bit29:28=2b11)
+	 * NS can access interrupts (bit27=1)
+	 * NS can lockown cache lines (bit26=1)
+	 * Pseudo-random replacement policy (bit25=0)
+	 * Force write allocated (default)
+	 * Shared attribute internally ignored (bit22=1, bit13=0)
+	 * Parity disabled (bit21=0)
+	 * Event monitor disabled (bit20=0)
+	 * Set or preserved way config: size (bit19:17), ass (bit16)
+	 * Store buffer device limitation enabled (bit11=1)
+	 * Cacheable accesses have high prio (bit10=0)
+	 * Full Line Zero (FLZ) disabled (bit0=0)
+	 */
+#ifdef PL310_PREVERSE_AUXCTRL_WAY_CONFIG
+#define PL310_WAYCFG_BITMASK	SHIFT_U32(0xF, 16)
```

Comment:
ohhh! i didn't catch it. anyway, I'll drop it. I will introduce it back if we get a large scale of soc variants we would like vanilla op-tee to support. In the mean time, we'll go for new flavors or local patches.